library/modules/Screen.cpp

```diff
@@ -956,6 +956,18 @@ int dfhack_lua_viewscreen::do_notify(lua_State *L)
     return 1;
 }
 
+void dfhack_lua_viewscreen::markInputAsHandled() {
+    if (!enabler)
+        return;
+
+    // clear text buffer
+    enabler->last_text_input[0] = '\0';
+
+    // mark clicked mouse buttons as handled
+    enabler->mouse_lbut = 0;
+    enabler->mouse_rbut = 0;
```

Comment:
Should this also handle mbut? or the _down fields?

Reply:
We don't touch mbut in general so that map scrolling continues to work. The _down fields should not be modified since it can break dragging. Widgets capture the mouse on button click and then react to _down if they handled the click. if we mess with _down, then we can break whatever thinks the mouse belongs to them (including our own window dragging mechanism).